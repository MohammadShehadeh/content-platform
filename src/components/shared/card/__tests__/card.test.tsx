import { render, screen } from '@testing-library/react';

import {
  Card,
  CardHeader,
  CardTitle,
  CardDescription,
  CardContent,
  CardFooter,
} from '@/components/shared/card';

describe('Card', () => {
  it('should render with children', () => {
    render(<Card>Hello</Card>);
    expect(screen.getByText('Hello')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Card>Hello</Card>);

    expect(container).toMatchSnapshot();
  });

  it('should render a complete card with all subcomponents', () => {
    const { container } = render(
      <Card className="test-card">
        <CardHeader>
          <CardTitle>Card Title</CardTitle>
          <CardDescription>This is a card description</CardDescription>
        </CardHeader>
        <CardContent>
          <p>This is the main content of the card</p>
        </CardContent>
        <CardFooter>
          <button>Action Button</button>
        </CardFooter>
      </Card>
    );

    expect(screen.getByText('Card Title')).toBeInTheDocument();
    expect(screen.getByText('This is a card description')).toBeInTheDocument();
    expect(screen.getByText('This is the main content of the card')).toBeInTheDocument();
    expect(screen.getByText('Action Button')).toBeInTheDocument();
    expect(container.firstChild).toHaveClass('test-card');
    expect(container).toMatchSnapshot();
  });
});
