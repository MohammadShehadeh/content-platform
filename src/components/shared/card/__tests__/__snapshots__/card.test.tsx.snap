// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<div>
  <div
    class="card"
  >
    Hello
  </div>
</div>
`;

exports[`Card renders a complete card with all subcomponents 1`] = `
<div>
  <div
    class="card test-card"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        Card Title
      </h3>
      <p
        class="description"
      >
        This is a card description
      </p>
    </div>
    <div
      class="content"
    >
      <p>
        This is the main content of the card
      </p>
    </div>
    <div
      class="footer"
    >
      <button>
        Action Button
      </button>
    </div>
  </div>
</div>
`;

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    Hello
  </div>
</div>
`;

exports[`Card should render a complete card with all subcomponents 1`] = `
<div>
  <div
    class="card test-card"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        Card Title
      </h3>
      <p
        class="description"
      >
        This is a card description
      </p>
    </div>
    <div
      class="content"
    >
      <p>
        This is the main content of the card
      </p>
    </div>
    <div
      class="footer"
    >
      <button>
        Action Button
      </button>
    </div>
  </div>
</div>
`;
